File: app/src/main/java/com/main/climbingdiary/models/MenuValues.java
package com.main.climbingdiary.models;

public enum MenuValues {
    SEARCH, FILTER, SORT, SORT_DATE, SETTINGS;

    public static MenuValues stringToSportType(String type) {
        return MenuValues.valueOf(type.toUpperCase());
    }

    public String toString() {
        String value = "";
        switch (this) {
            case SEARCH:
                value = "search";
                break;
            case FILTER:
                value = "filter";
                break;
            case SORT:
                value = "sort";
                break;
            case SORT_DATE:
                value = "date";
                break;
            case SETTINGS:
                value = "settings";
                break;
        }
        return value;
    }

    public String typeToString() {
        return this.toString().toLowerCase();
    }
}

File: app/src/main/java/com/main/climbingdiary/models/MenuValues.kt
package com.main.climbingdiary.models

import java.util.*


enum class MenuValues {
    SEARCH, FILTER, SORT, SORT_DATE, SETTINGS;

    companion object {
        fun stringToSportType(type: String?): MenuValues {
            return valueOf(type!!.toUpperCase(Locale.ROOT))
        }
    }

    fun typeToString(): String {
        return this.toString().toLowerCase(Locale.ROOT)
    }

    override fun toString(): String {
        var value = ""
        value = when (this) {
            SEARCH -> "search"
            FILTER -> "filter"
            SORT -> "sort"
            SORT_DATE -> "date"
            SETTINGS -> "settings"
        }
        return value
    }
}